File: rebate/src/test/java/com/binance4j/rebate/client/RebateClientTest.java
package com.binance4j.rebate.client;

import java.util.List;

import org.junit.jupiter.api.Test;

import com.binance4j.core.exception.ApiException;
import com.binance4j.core.param.FramedPaging;
import com.binance4j.core.test.CustomTest;
import com.fasterxml.jackson.databind.DeserializationFeature;

public class RebateClientTest extends CustomTest {
	RebateClient client = new RebateClient(key, secret);

	public RebateClientTest() {
		client.getMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, true);
		client.getMapper().configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, true);
	}

	@Test
	void testGetSpotRebateHistoryRecords() throws ApiException {
		client.getMapper().configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, false);
		testHasNulls(client.getSpotRebateHistoryRecords(), List.of("data"), true);
		client.getMapper().configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, true);
	}

	@Test
	void testGetSpotRebateHistoryRecords2() throws ApiException {
		client.getMapper().configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, false);
		testHasNulls(client.getSpotRebateHistoryRecords(new FramedPaging(1)), List.of("data"), true);
		client.getMapper().configure(DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES, true);
	}
}

File: src/test/com/binance4j/connectors/rebate/client/RebateClientTest.kt
package com.binance4j.connectors.rebate.client

import com.binance4j.connectors.Connectors
import com.binance4j.connectors.core.test.CustomTest
import org.junit.jupiter.api.Test

internal class RebateClientTest : CustomTest() {
    private var client = Connectors.rest(key, secret).rebate

    @Test
    fun `Get All Spot Rebate History Records`() = assertDoesNotThrow(client.getSpotRebateHistoryRecords())

    @Test
    fun `Get All Spot Rebate History Records in an interval`() = assertDoesNotThrow(client.getSpotRebateHistoryRecords(lastHour, today, 1, limit))
}